src/event_loop.rs

```diff
@@ -462,9 +472,31 @@ impl<T: 'static> EventLoopProxy<T> {
     /// Returns an `Err` if the associated [`EventLoop`] no longer exists.
     ///
     /// [`UserEvent(event)`]: Event::UserEvent
+    #[deprecated = "get a `waker` and wake that instead"]
     pub fn send_event(&self, event: T) -> Result<(), EventLoopClosed<T>> {
         self.event_loop_proxy.send_event(event)
     }
+
+    /// Create a [`Waker`] that can wake the event loop from any thread.
+    ///
+    /// This is usually used in combination with [`mpsc::channel`] to send the
+    /// data that should be processed by the event loop.
+    ///
+    /// If you do use a channel, the data will be accessible from inside
+    /// [`Event::NewEvents`].
+    ///
+    /// [`mpsc::channel`]: std::sync::mpsc::channel
+    pub fn waker(self) -> Waker {
+        self.event_loop_proxy.waker()
+    }
+}
+
+// Note: This does not have the generic from `EventLoopProxy<T>`, since this
+// is the new API that shouldn't need it.
+impl From<EventLoopProxy> for Waker {
+    fn from(proxy: EventLoopProxy) -> Self {
+        proxy.waker()
+    }
 }
```

Comment:
I'm not sure this is wanted — eventually `EventLoopProxy` will (presumably) just be replaced with a `Waker`.

In fact, maybe just deprecate `EventLoopProxy` and `create_proxy` now, replacing with `EventLoop::create_waker`?

Reply:
I want to reuse proxy in the future for some internal stuff in the future around `Send`/`Sync` and also, not be limited on the `Waker` API, since it's rather limited. But yeah, in the current state they are mostly the same.